### ISSUE TITLE
Fails to build on 14-.0-CURRENT

### ISSUE THREAD

**Issue by koobs** [KlaraSystems/kvmclock#2]
## Summary

0.0.0-a4 (via sysutils/kvmclock) fails to build on FreeBSD 14.0-CURRENT

## uname -a

FreeBSD 14.0-CURRENT #2 main-n249859-248682a5891: Mon Oct  4 03:20:27 UTC 2021     root@140-CURRENT-amd64-564d:/usr/obj/usr/src/amd64.amd64/sys/GENERIC-NODEBUG  amd64

## Error log

```
--- rdtsc_ordered.o ---
cc  -march=sandybridge -mtune=sandybridge -O2 -pipe -fcolor-diagnostics -DINVARIANTS -DINVARIANT_SUPPORT -march=sandybridge  -fno-strict-aliasing -D_KERNEL -DKLD_MODULE -nostdinc  -Iinclude -include /var/tmp/tmpfs0/usr/home/koobs/repos/freebsd/ports-git/sysutils/kvmclock-kmod/work/kvmclock-0.0.0-a4/opt_global.h -I. -I/usr/src/sys -I/usr/src/sys/contrib/ck/include -fno-common  -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fdebug-prefix-map=./machine=/usr/src/sys/amd64/include -fdebug-prefix-map=./x86=/usr/src/sys/x86/include     -MD  -MF.depend.rdtsc_ordered.o -MTrdtsc_ordered.o -mcmodel=kernel -mno-red-zone -mno-mmx -mno-sse -msoft-float  -fno-asynchronous-unwind-tables -ffreestanding -fwrapv -fstack-protector -Wall -Wredundant-decls -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith -Wcast-qual -Wundef -Wno-pointer-sign -D__printf__=__freebsd_kprintf__ -Wmissing-include-dirs -fdiagnostics-show-option -Wno-unknown-pragmas -Wno-error=tautological-compare -Wno-error=empty-body -Wno-error=parentheses-equality -Wno-error=unused-function -Wno-error=pointer-sign -Wno-error=shift-negative-value -Wno-address-of-packed-member -Wno-format-zero-length   -mno-aes -mno-avx  -std=iso9899:1999 -c rdtsc_ordered.c -o rdtsc_ordered.o
rdtsc_ordered.c:34:1: error: redefinition of 'rdtsc_ordered_lfence'
DEFINE_RDTSC_ORDERED();
^
include/x86/_rdtsc_ordered.h:84:5: note: expanded from macro 'DEFINE_RDTSC_ORDERED'
    DEFINE_RDTSC_ORDERED_COMMON()                                       \
    ^
include/x86/_rdtsc_ordered.h:52:5: note: expanded from macro 'DEFINE_RDTSC_ORDERED_COMMON'
    rdtsc_ordered_lfence(void)                                          \
    ^
./machine/cpufunc.h:359:1: note: previous definition is here
rdtsc_ordered_lfence(void)
^
rdtsc_ordered.c:34:1: error: redefinition of 'rdtsc_ordered_mfence'
DEFINE_RDTSC_ORDERED();
^
include/x86/_rdtsc_ordered.h:84:5: note: expanded from macro 'DEFINE_RDTSC_ORDERED'
    DEFINE_RDTSC_ORDERED_COMMON()                                       \
    ^
include/x86/_rdtsc_ordered.h:58:5: note: expanded from macro 'DEFINE_RDTSC_ORDERED_COMMON'
    rdtsc_ordered_mfence(void)                                          \
    ^
./machine/cpufunc.h:366:1: note: previous definition is here
rdtsc_ordered_mfence(void)
^
2 errors generated.
*** [rdtsc_ordered.o] Error code 1

make[2]: stopped in /var/tmp/tmpfs0/usr/home/koobs/repos/freebsd/ports-git/sysutils/kvmclock-kmod/work/kvmclock-0.0.0-a4
```

## /etc/make.conf

```
# OPTIMIZE ALL THE THINGS

CFLAGS=         -march=sandybridge -mtune=sandybridge -O2 -pipe -fcolor-diagnostics
CPUTYPE?=       sandybridge

.if !empty(.CURDIR:M/*/ports*/*)
SSP_CFLAGS=     -fstack-protector-strong
USE_LTO=        yes
LTO_TOOLCHAIN=  AR=llvm-ar \
                RANLIB=llvm-ranlib \
                NM=llvm-nm \
                OBJDUMP=llvm-objdump

.if defined(USE_LTO)
AR=     llvm-ar
RANLIB= llvm-ranlib
NM=     llvm-nm
OBJDUMP=llvm-objdump
LTO_TOOLCHAIN=  AR=${AR} \
                RANLIB=${RANLIB} \
                NM=${RANLIB} \
                OBJDUMP=#{OBJDUMP}

#CONFIGURE_ARGS+=       ${LTO_TOOLCHAIN}
MAKE_ENV+=              ${LTO_TOOLCHAIN}
#MAKE_ARGS+=            ${LTO_TOOLCHAIN}
.endif
.endif
```

## /etc/src.conf

```
# Speed
MALLOC_PRODUCTION=yes
WITHOUT_LLVM_ASSERTIONS=ses

# Hardening
WITH_PIE=yes
WITH_BIND_NOW=yes
WITH_RETPOLINE=yes
WITH_SSP=yes

# Sanitizers
#WITH_ASAN=yes
#WITH_UBSAN=yes

WITH_CCACHE_BUILD=yes
CCACHE_PREFIX=/usr/local/bin/distcc
```

**Comment by koobs** [comment]
`0.0.0.a5` builds on `14.0-CURRENT main-n256631-16e4487e5f8 GENERIC-NODEBUG amd64`

Closing resolved

**Comment by allanjude** [comment]
Note: the fuller-featured version of the kvmclock drive is included in 14.0 and 13.1, so this port can be retired once 12.x is EoL
